script.js

```diff
@@ -1,4 +1,184 @@
+After both players have rolled and chosen dice order, the player with the higher combined number wins.*/
+
+// -1: Introduction and game mode selection
+// 0 : Before player 1 roll dice
+// 1 : After player 1 roll dice
+// 2 : Before player 2 roll dice
+// 3 : After player 2 roll dice
+var turnOrOrder = -1;
+
+// 1: Normal mode
+// 2: Reversed Mode
+var gameMode = 0;
+
+// 0 : No action needed
+// 1 : Action needed
+var whichDiceOrder = 0;
+
+// to track overall results
+var pOneStats = 0;
+var pTwoStats = 0;
+var leaderboard = "";
+var myOutputValue;
+
+var playerThrow = [0, 0];
+var pOneResult = 0;
+var pTwoResult = 0;
+
+// Dice roll function to get random dice number
+var diceRollResult = function () {
+  var rollResult = Math.floor(Math.random() * 6) + 1;
+  return rollResult;
+};
+
+// Evaluate results
+var evaluateResults = function (input) {
+  //For both player first roll
+  if ((turnOrOrder == 0 || turnOrOrder == 2) && whichDiceOrder == 0) {
+    whichDiceOrder = 1;
+    turnOrOrder += 1;
+    playerThrow[0] = diceRollResult();
+    playerThrow[1] = diceRollResult();
+    myOutputValue = `You rolled ${playerThrow[0]} and ${playerThrow[1]}.`;
+    console.log(turnOrOrder);
+
+    //If both die result is the same, skip order selection
+    if (playerThrow[0] == playerThrow[1]) {
+      whichDiceOrder = 0;
+      turnOrOrder += 1;
+      //result for P1
+      if (turnOrOrder == 2) {
+        pOneResult = "" + playerThrow[0] + playerThrow[1];
+        //result for P2
+        myOutputValue =
+          myOutputValue +
+          `<br><br>Your hand is ${playerThrow[0]}${playerThrow[1]}.`;
+        return myOutputValue;
+      } else if (turnOrOrder == 4) {
+        pTwoResult = "" + playerThrow[0] + playerThrow[1];
+        myOutputValue =
+          myOutputValue +
+          `<br><br>Your hand is ${playerThrow[0]}${playerThrow[1]}.`;
+        return myOutputValue;
+      }
+    }
+
+    //Else ask player which order they want to choose
+    myOutputValue =
+      myOutputValue +
+      ` <br><br>Which order would you like to put them?<br> 1) die 1 first : ${playerThrow[0]}${playerThrow[1]} <br> 2) die 2 first : ${playerThrow[1]}${playerThrow[0]}`;
+    return myOutputValue;
+  } // Player to choose order of die:
+  else if (turnOrOrder == 1 && whichDiceOrder == 1) {
+    // Player 1 return message (die 1 first)
+    if (input == 1) {
+      turnOrOrder += 1;
+      whichDiceOrder = 0;
+      pOneResult = "" + playerThrow[0] + playerThrow[1];
+      return `Ok, your hand is ${pOneResult}. <br><br>It is now Player 2's turn.`;
+    } // player 1 return message (die 2 first)
+    else if (input == 2) {
+      turnOrOrder += 1;
+      whichDiceOrder = 0;
+      pOneResult = "" + playerThrow[1] + playerThrow[0];
+      return `Ok, your hand is ${pOneResult}. <br><br>It is now Player 2's turn.`;
+    } else {
+      return `Please select a valid number.<br><br>${myOutputValue}`;
+    }
```

Comment:
commenting this here but this actually applies to the bulk of your code. you mentioned that a specific feedback you want is how to visualise nested `if`s. i did notice that your code has many layers of nesting and one thing I emphasised since the start is making use of early return, which you are doing. but when we return early there is usually no need to nest the rest of your code in additional `else if`, `else`, etc. 

An example for this little section here:
```
    if (input == 1) {
      turnOrOrder += 1;
      whichDiceOrder = 0;
      pOneResult = "" + playerThrow[0] + playerThrow[1];
      return `Ok, your hand is ${pOneResult}. <br><br>It is now Player 2's turn.`;
    } // player 1 return message (die 2 first)
    
    if (input == 2) {
      turnOrOrder += 1;
      whichDiceOrder = 0;
      pOneResult = "" + playerThrow[1] + playerThrow[0];
      return `Ok, your hand is ${pOneResult}. <br><br>It is now Player 2's turn.`;
    } 

    return `Please select a valid number.<br><br>${myOutputValue}`;
```

The way the code above works is: if the first condition is met return early, otherwise it carries on to the second condition (which is why we don't need the else if), again if it is met then we return early. Otherwise we carry on to the last part which does not need to be in a conditional statement, this serves as our 'default' case if we didn't return early.

By applying this fix to other parts of your code, I'm sure it will reduce your nested `if` drastically which will eliminate the problem of having trouble visualising the layers of nesting.